pkgs/applications/networking/mullvad-vpn/default.nix

```
@@ -1,145 +1,81 @@
-    inherit hash;
-  };
+  npmFlags = [ "--ignore-scripts" ];
 
   nativeBuildInputs = [
-    autoPatchelfHook
-    dpkg
     makeWrapper
+    protoc-gen-js
   ];
 
-  buildInputs = deps;
-
-  dontBuild = true;
-  dontConfigure = true;
+  env = {
+    ELECTRON_SKIP_BINARY_DOWNLOAD = "1";
+    ELECTRON_OVERRIDE_DIST_PATH = electron.dist; # TODO: figure out why this is needed
+  };
 
-  unpackPhase = "dpkg-deb -x $src .";
+  postConfigure = ''
+    ln -s ${lib.getBin mullvad}/bin/* dist-assets/
+    ln -s ${lib.getExe' protobuf "protoc"} gui/node_modules/grpc-tools/bin/protoc
+    ln -s ${lib.getExe' grpc-tools "grpc_node_plugin"} gui/node_modules/grpc-tools/bin/grpc_node_plugin
+  '';
 
-  runtimeDependencies = [ (lib.getLib systemd) libGL libnotify libappindicator wayland ];
+  preBuild = "pushd gui";
+  npmBuildScript = "pack:linux";
+  postBuild = "popd";
 
   installPhase = ''
     runHook preInstall
 
-    mkdir -p $out/share/mullvad $out/bin
-
-    mv usr/share/* $out/share
-    mv usr/bin/* $out/bin
-    mv opt/Mullvad\ VPN/* $out/share/mullvad
-
-    ln -s $out/share/mullvad/mullvad-{gui,vpn} $out/bin/
-    ln -sf $out/share/mullvad/resources/mullvad-problem-report $out/bin/mullvad-problem-report
-
-    wrapProgram $out/bin/mullvad-vpn \
-      --set MULLVAD_DISABLE_UPDATE_NOTIFICATION 1 \
-      --prefix PATH : ${lib.makeBinPath [ coreutils gnugrep ]}
+    mkdir -p $out/share/mullvadvpn-gui/
+    cp -r dist/*-unpacked/{locales,resources{,.pak}} $out/share/mullvadvpn-gui/
 
-    wrapProgram $out/bin/mullvad-daemon \
-        --set-default MULLVAD_RESOURCE_DIR "$out/share/mullvad/resources"
-
-    wrapProgram $out/bin/mullvad-gui \
-      --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--enable-features=UseOzonePlatform --ozone-platform=wayland}}"
-
-    sed -i "s|Exec.*$|Exec=$out/bin/mullvad-vpn $U|" $out/share/applications/mullvad-vpn.desktop
+    makeWrapper ${lib.getExe electron} $out/bin/mullvadvpn-gui \
+        --add-flags $out/share/mullvadvpn-gui/resources/app.asar \
+        --set MULLVAD_DISABLE_UPDATE_NOTIFICATION 1 \
+        --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}}" \
+        --inherit-argv0
 
     runHook postInstall
   '';
 
-  passthru.updateScript = ./update.sh;
-
-  meta = with lib; {
+  meta = {
     homepage = "https://github.com/mullvad/mullvadvpn-app";
     description = "Client for Mullvad VPN";
-    changelog = "https://github.com/mullvad/mullvadvpn-app/blob/${version}/CHANGELOG.md";
-    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
-    license = licenses.gpl3Only;
-    platforms = [ "x86_64-linux" "aarch64-linux" ];
-    maintainers = with maintainers; [ Br1ght0ne ymarkus ataraxiasjel ];
+    changelog = "https://github.com/mullvad/mullvadvpn-app/blob/${src.rev}/CHANGELOG.md";
+    license = lib.licenses.gpl3Only;
+    platforms = [
+      "x86_64-linux"
+      "aarch64-linux"
+    ];
```

```suggestion
    platforms = lib.platforms.unix;
```
Since we're now building from source, it's possible that more platforms will work.

---

A package should restrict platforms based on what the package itself could theoretically support if all of its dependencies worked as intended on the platform. If one of those dependencies did not work, eval would simply fail at that dependency.

I don't see any need to restrict platforms for this package. Simply leave it at the default which is `all`.

```suggestion
```

If the build is broken on i.e. Darwin but it should work in theory, you should set meta.broken accordingly, ideally with a hint as to why it doesn't work. Again though, if it's broken due to a broken dep, only that dep should be marked as broken and that will make the eval fail.

---

Thanks, I'll just set it to `electron.meta.platforms`.

(I might take a look at which dependency fails on darwin to try to fix it)